pil-analyzer/src/expression_processor.rs

```diff
@@ -362,3 +381,17 @@ struct LocalVariableState {
     pub local_variables: HashMap<String, u64>,
     pub local_variable_counter: u64,
 }
+
+fn combine_source_refs(refs: Vec<&SourceRef>) -> SourceRef {
```

Comment:
Maybe this was not such a good suggestion of mine. What happens if the source refs are from different files? Maybe it's best to keep it at Unknown. Sorry!